torchgeo/datamodules/flair2.py

```diff
@@ -0,0 +1,85 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+#
+# FLAIR dataset is released under open license 2.0
+# ..... https://www.etalab.gouv.fr/wp-content/uploads/2018/11/open-licence.pdf
+# ..... https://ignf.github.io/FLAIR/#FLAIR2
+
+"""This module contains the FLAIR2DataModule class for loading the FLAIR2 dataset.
+
+The FLAIR dataset is released under open license 2.0:
+- https://www.etalab.gouv.fr/wp-content/uploads/2018/11/open-licence.pdf
+- https://ignf.github.io/FLAIR/#FLAIR2
+
+Code for loading dataset licensed under the MIT License.
+"""
+
+from typing import Any
+
+import kornia.augmentation as K
+import torch
+from torch.utils.data import random_split
+
+from ..datasets import FLAIR2, FLAIR2Toy
+from ..samplers.utils import _to_tuple
+from ..transforms import AugmentationSequential
+from .geo import NonGeoDataModule
+
+
+class FLAIR2DataModule(NonGeoDataModule):
+    """LightningDataModule implementation for the FLAIR2 dataset.
+
+    Uses the train/test splits from the dataset.
+    """
+
+    def __init__(
+        self,
+        batch_size: int = 64,
+        patch_size: tuple[int, int] | int = 64,
+        val_split_pct: float = 0.2,
+        num_workers: int = 0,
+        use_toy: bool = False,
+        augs: AugmentationSequential | None = None,
+        **kwargs: Any,
+    ) -> None:
+        """Initialize a new FLAIR2DataModule instance.
+
+        Args:
+            batch_size: Size of each mini-batch.
+            patch_size: Size of each patch, either ``size`` or ``(height, width)``.
+                Should be a multiple of 32 for most segmentation architectures.
+            val_split_pct: Percentage of the dataset to use as a validation set.
+            num_workers: Number of workers for parallel data loading.
+            use_toy: Whether to use the toy version of the dataset.
+            augs: Optional augmentations to apply to the dataset.
```

Comment:
We have an ongoing debate on how best to make augmentations customizable. For now, I would avoid adding this until that debate settles down.